src/app/home/home.component.ts

```diff
@@ -222,36 +244,49 @@ export class HomeComponent implements AfterViewInit {
     }
 
     // TODO there might be a bug here that creates double spaces in the text, test more
-    deleteTextMarking(index: number): void {
+    /**
+     * Delete the TextMarking based on the textMarkingIndex.
+     * @param textMarkingIndex {number}
+     */
+    deleteTextMarking(textMarkingIndex: number): void {
         if (this.displayWriteTextOrUploadDocumentFlag) {
             const editor = document.getElementById(this.EDITOR_KEY);
             const htmlElement: HTMLBodyElement = new DOMParser().parseFromString(editor!.innerHTML, "text/html")
                 .firstChild!.lastChild! as HTMLBodyElement;
 
-            htmlElement.replaceChild(document.createTextNode(htmlElement.children[index].textContent!), htmlElement.children[index]);
+            htmlElement.replaceChild(document.createTextNode(htmlElement.children[textMarkingIndex].textContent!), htmlElement.children[textMarkingIndex]);
 
             editor!.innerHTML = htmlElement.innerHTML;
         }
 
         this.processedText!.textMarkings = this.processedText!.textMarkings
-            .filter(tM => tM != this.processedText!.textMarkings[index]);
+            .filter(tM => tM != this.processedText!.textMarkings[textMarkingIndex]);
         this.shouldCollapseSuggestions = new Array<boolean>(this.processedText!.textMarkings.length).fill(true);
     }
 
-    saveSelection(elementNode: any) {
-        const range = window.getSelection()!.getRangeAt(0);
-        const preSelectionRange = range.cloneRange();
+    /**
+     * Save the position of the current selection based on a start and end number.
+     * @param elementNode
+     */
+    saveSelection(elementNode: any): { start: number, end: number } {
+        const range: Range = window.getSelection()!.getRangeAt(0);
+        const preSelectionRange: Range = range.cloneRange();
         preSelectionRange.selectNodeContents(elementNode);
         preSelectionRange.setEnd(range.startContainer, range.startOffset);
-        const start = preSelectionRange.toString().length;
+        const start: number = preSelectionRange.toString().length;
 
         return {
             start: start,
             end: start + range.toString().length
         }
     };
 
-    restoreSelection(elementNode: any, savedSelection: any) {
+    /**
+     * Restore the selection to an earlier save.
+     * @param elementNode
+     * @param savedSelection
```

Comment:
I would give a suggestion here, but I don't understand this myself. what's a "save". Is it a saved state or of which/where. Where is it being saved at.
In addition to that, adding descriptions to the two parameters below here would help

Reply:
Should be now more detailed